src/Tizen.Sensor/Tizen.Sensor/Sensor.cs

```diff
@@ -59,9 +181,26 @@ internal Sensor(uint index)
             {
                 CreateListener();
                 GetProperty();
+                RecorderCreateOption();
+                RecorderCreateQuery();
+                CreateProvider(_uri);
             }
         }
 
+        internal Sensor(String uri, uint index)
+        {
+            GetHandleList(uri, index);
+            if (CheckSensorHandle())
+            {
+                CreateListener();
+                GetProperty();
+                RecorderCreateOption();
+                RecorderCreateQuery();
+                CreateProvider(uri);
+            }
```

Comment:
Extract method from 193 to 197 lines
and use it in both constructor

``` c#
internal Sensor(String uri, uint index)
{
    GetHandleList(uri, index);
    InitializeSensor();
}
void InitializeSensor() {
    if (CheckSensorHandle())
    {
        CreateListener();
        GetProperty();
        RecorderCreateOption();
        RecorderCreateQuery();
        CreateProvider(uri);
    }
}
```

Reply:
Created new provider and recorder classes so removed all the API's related to them from sensor.cs

Reply:
```
            if (CheckSensorHandle())
            {
                CreateListener();
                GetProperty();
            }
```
this code is duplicated in both constructor. So you need to extract method